### PR TITLE
Fix logging

### PR DESCRIPTION
**Pull request by astoycos**
Try and fixup how are logging is organized a works

This is now the logs from a load and unload of two separate programs with `RUST_LOG=bpfd=info`.

```bash
---------------->>>>>>>>> STARTUP
[astoycos@nfvsdn-02-oot bpfd]$ sudo RUST_LOG=bpfd=info ./target/debug/bpfd
[INFO  bpfd] Log using env_logger
[INFO  bpfd] Has CAP_BPF: true
[INFO  bpfd] Has CAP_SYS_ADMIN: true
[WARN  bpfd] Unable to read config file, using defaults
[INFO  bpfd::certs] CA Certificate file /etc/bpfd/certs/ca/ca.pem does not exist. Creating CA Certificate.
[INFO  bpfd::certs] bpfd Certificate Key /etc/bpfd/certs/bpfd/bpfd.key does not exist. Creating bpfd Certificate.
[INFO  bpfd::certs] bpfd-client Certificate Key /etc/bpfd/certs/bpfd-client/bpfd-client.key does not exist. Creating bpfd-client Certificate.
[INFO  bpfd::oci_utils::cosign] Starting Cosign Verifier, downloading data from Sigstore TUF repository
[INFO  bpfd::serve] Listening on [::1]:50051
---------------->>>>>>>>> LOAD 1
[WARN  bpfd::oci_utils::cosign] The bytecode image: quay.io/bpfd-bytecode/xdp_pass:latest is unsigned
[INFO  bpfd::command] Loading program bytecode from container image: quay.io/bpfd-bytecode/xdp_pass:latest
[INFO  bpfd::bpf] Adding xdp program: pass
[INFO  bpfd::bpf] Removing program with id: 30845
---------------->>>>>>>>> LOAD 2
[WARN  bpfd::oci_utils::cosign] The bytecode image: quay.io/bpfd-bytecode/uprobe:latest is unsigned
[INFO  bpfd::command] Loading program bytecode from container image: quay.io/bpfd-bytecode/uprobe:latest
[INFO  bpfd::bpf] Adding probe program: my_uprobe
[INFO  bpfd::bpf] Removing program with id: 30846

```
All other commands intentionally don't log when at the info level. 

Additionally fix-up how we get and store the raw program bytes during the load process

### CODE REVIEW

File: bpfd/src/bpf.rs [hunk 1]
```diff
@@ -284,6 +282,13 @@ impl BpfManager {
             .data_mut()?
             .set_program_bytes(self.image_manager.clone())
             .await?;
+
+        info!(
```
[the remark this answers is not in the document]
Comment (astoycos) [reply bpfd/src/bpf.rs:286]: Done, and I don't think so... since we're propagating any errors back to the user in the rpc status field.

**Comment by astoycos** [comment]
@Billy99 I'm happy to add more here but I think this is a pretty good stopgap for our release....LMK what you think 

The k8s logging is now always defaulted to info for both the agent and bpfd

File: bpfd-operator/config/bpfd-deployment/config.yaml [hunk 2]
```diff
@@ -7,7 +7,10 @@ data:
   ## Can be configured at runtime
   bpfd.agent.image: quay.io/bpfd/bpfd-agent:latest
   bpfd.image: quay.io/bpfd/bpfd:latest
-  bpfd.log.level: "info"
+  ## Can be set to "info", "debug", or "trace"
```
Comment (Billy99) [inline bpfd-operator/config/bpfd-deployment/config.yaml:10]: Update the docs.
docs/developer-guide/logging.md

File: bpfd-operator/config/bpfd-deployment/config.yaml [hunk 3]
```diff
@@ -7,7 +7,10 @@ data:
...
+  bpfd.agent.log.level: "info"
+  ## See https://docs.rs/env_logger/latest/env_logger/ for configuration options
+  bpfd.log.level: "bpfd=info"
```
Comment (Billy99) [inline bpfd-operator/config/bpfd-deployment/config.yaml:13]: Is there a reason they aren't consistent?
```
bpfd.agent.log.level: "info"
bpfd.bpfd.log.level: "info"
```
OR
```
bpfd.log.level: "bpfd=info,agent=info"
```

**Review by Billy99 — CHANGES_REQUESTED** [review]

File: bpfd-operator/config/bpfd-deployment/config.yaml (hunk 3 above)
Comment (astoycos) [reply bpfd-operator/config/bpfd-deployment/config.yaml]: So with "bpfd=info" I was specifying to the rust process ONLY log for the bpfd crate at the info level.

I changed it for now to just be info....but to cut down on some of the dependency log verbosities when trying to debug it's often useful to do `bpfd=debug` for `bpfd.log.level`

However I want to leave the logging fully configurable to help users pinpoint problems (i.e `aya=debug` is also a valid configuration option)

File: bpfd-operator/config/bpfd-deployment/config.yaml (hunk 2 above)
Comment (astoycos) [reply bpfd-operator/config/bpfd-deployment/config.yaml:10]: Done!

File: bpfd/src/bpf.rs [hunk 4]
```diff
@@ -782,7 +787,8 @@ impl BpfManager {
     }
 
     pub(crate) fn get_program(&mut self, id: u32) -> Result<Program, BpfdError> {
-        debug!("BpfManager::get_program({0})", id);
+        debug!("BpfManager::get_program()");
```
[the remark this answers is not in the document]
Comment (astoycos) [reply bpfd/src/bpf.rs]: Done!

File: bpfd/src/bpf.rs [hunk 5]
```diff
@@ -280,37 +278,46 @@ impl BpfManager {
...
 
+        // Program bytes MUST be cleared after load.
+        program.data_mut()?.clear_program_bytes();
+
         // map_pin_path MUST be set following load.
         let map_pin_path = program.data()?.map_pin_path();
 
         match result {
             Ok(id) => {
+                info!(
```
[the remark this answers is not in the document]
Comment (astoycos) [reply bpfd/src/bpf.rs:306]: Done!

File: bpfd-operator/config/bpfd-deployment/config.yaml (hunk 3 above)
Comment (Billy99) [reply bpfd-operator/config/bpfd-deployment/config.yaml]: I'm going to go ahead and merge it, but the `aya=debug` should probably be in the docs too.

**Review by Billy99 — APPROVED** [review]
LGTM

**Comment by Billy99** [comment]
Resolves: #663

**Pull request merged**